internal/git/gitsync_processor.go

```diff
@@ -96,15 +116,177 @@ func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ s
 		logger.Debug("no monitoring")
 
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				changedFiles, err := CheckForRepoUpdates(r, repo, &processor.gitSync.Status, ctx)
+				if err != nil {
+					return err
+				}
+				// apply the resources which are updated
+				for _, v := range changedFiles.ModifiedFiles {
+					// split the string on the basis of "---"
+					individualResource := strings.Split(string(v), "---")
```

Comment:
cool. I guess we'll need to see if ultimately we also need to remove any empty lines that surround the "---" in order to pass to Chandan's method

Reply:
So, here we're going to need to determine which of the resources have been deleted within the file, right?

Reply:
I have considered a scenario here when the particular file has been removed from the git ,that files resources will be deleted . I guess you are right here ,i have to check for the what content has been removed from the files too if they are updated ,but I am not sure how it can be done ,do I have to read the whole file before and after and then compare the contents ?  please suggest